File: Triatlon-Java/ChatPersistence/src/main/java/triathlon/persistence/repository/jdbc/triatlon/ResultDBRepository.java
package triathlon.persistence.repository.jdbc.triatlon;

import triathlon.model.Result;
import triathlon.model.Participant;
import triathlon.model.Stage;


import triathlon.persistence.ResultRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;


import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class ResultDBRepository implements ResultRepository {
    private JdbcUtils dbUtils;

    private static final Logger logger = LogManager.getLogger();

    public ResultDBRepository(Properties properties) {
        logger.info("Initializing ResultDBRepository with properties: {}", properties);
        this.dbUtils = new JdbcUtils(properties);
    }

    @Override
    public Result findOne(Integer integer) {
        Result searchedResult = null;
        logger.traceEntry();
        Connection connection = dbUtils.getConnection();

        try (PreparedStatement preparedStatement = connection.prepareStatement("select * from Results where id = ?")){
            preparedStatement.setInt(1, integer);
            try (ResultSet resultSet = preparedStatement.executeQuery()){
                while (resultSet.next()) {
                    int id = resultSet.getInt("id");
                    int idParticipant = resultSet.getInt("idParticipant");
                    int idStage = resultSet.getInt("idStage");
                    double score = resultSet.getDouble("score");

                    Participant participant = new Participant();
                    participant.setId(idParticipant);

                    Stage stage = new Stage();
                    stage.setId(idStage);

                    Result result = new Result(participant, stage, score);

                    searchedResult = result;
                    return result;
                }
            }
        }
        catch (SQLException ex) {
            logger.error(ex);
            System.err.println(ex);
        }

        return searchedResult;
    }

    @Override
    public Iterable<Result> findAll() {
        return null;
    }

    @Override
    public Result save(Result entity) {
        logger.traceEntry();
        Connection connection = dbUtils.getConnection();

        try (PreparedStatement preparedStatement = connection.prepareStatement("insert into Results (idParticipant, idStage, score) values(?,?,?)")){
            preparedStatement.setInt(1, entity.getParticipant().getId());
            preparedStatement.setInt(2, entity.getStage().getId());
            preparedStatement.setDouble(3, entity.getScore());

            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} results", result);
        }
        catch (SQLException ex) {
            logger.error(ex);
            System.err.println(ex);
        }
        logger.traceExit();
        return null;
    }

    @Override
    public Result delete(Integer integer) {
        return null;
    }

    @Override
    public Result update(Result entity) {
        return null;
    }
}

File: Triatlon-CS/Persistence/ResultDBRepository.cs
using System.Collections.Generic;
using System.Data;
using log4net;
using Triatlon_CSharp.Domain;
using Triatlon_CSharp.Utils;

namespace Triatlon_CSharp.Repository
{
    public class ResultDBRepository : ResultRepository
    {
        private static readonly ILog Log = LogManager.GetLogger("ResultDBRepository");

        public ResultDBRepository()
        {
            Log.Info("Creating ResultDBRepository");    
        }
        
        public Result FindOne(int id)
        {
            Result searchedResult = null;
            Log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection connection = DBUtils.getConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select * from Results where id=@id";
                IDbDataParameter paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                command.Parameters.Add(paramId);

                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        int idR = dataReader.GetInt32(0);
                        int idParticipant = dataReader.GetInt32(1);
                        int idStage = dataReader.GetInt32(2);
                        double score = dataReader.GetDouble(3);

                        Participant participant = new Participant();
                        participant.Id = idParticipant;
                        Stage stage = new Stage();
                        stage.Id = idStage;

                        Result result = new Result(participant, stage, score);
                        result.Id = idR;

                        searchedResult = result;
                    }
                }
            }

            return searchedResult;
        }

        public IEnumerable<Result> FindAll()
        {
            throw new System.NotImplementedException();
        }

        public Result Save(Result entity)
        {
            var connection = DBUtils.getConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "insert into Results (idParticipant, idStage, score) values (@idParticipant, @idStage, @score)";

                var paramIdParticipant = command.CreateParameter();
                paramIdParticipant.ParameterName = "@idParticipant";
                paramIdParticipant.Value = entity.Participant.Id;
                command.Parameters.Add(paramIdParticipant);
                
                var paramIdStage = command.CreateParameter();
                paramIdStage.ParameterName = "@idStage";
                paramIdStage.Value = entity.Stage.Id;
                command.Parameters.Add(paramIdStage);
                
                var paramScore = command.CreateParameter();
                paramScore.ParameterName = "@score";
                paramScore.Value = entity.Score;
                command.Parameters.Add(paramScore);

                var result = command.ExecuteNonQuery();
            }

            return null;
        }

        public Result Delete(int id)
        {
            throw new System.NotImplementedException();
        }

        public Result Update(Result entity)
        {
            throw new System.NotImplementedException();
        }
    }
}